### PR TITLE
Validation find unexpected elements

### LINKED ISSUE #942: Validator should throw warning if there are extra fields
As per hackathon discussion. 

Should exit with zero status with warnings only.

### PR DESCRIPTION
**Pull request by t-b**
Requires https://github.com/hdmf-dev/hdmf/pull/288

Close #942.

### CODE REVIEW

File: src/pynwb/validate.py [hunk 1]
```diff
@@ -45,13 +49,12 @@ def main():
 
     feature_parser = parser.add_mutually_exclusive_group(required=False)
     feature_parser.add_argument("--cached-namespace", dest="cached_namespace", action='store_true',
-                                help="Use the cached namespace (default).")
+            help="Use the cached namespace (default: %(default)s).", default=True)
```
Comment (rly) [inline src/pynwb/validate.py:52]:
```suggestion
            help="Use the cached namespace (default: %(default)).", default=True)
```

File: src/pynwb/validate.py [hunk 2]
```diff
@@ -45,13 +49,12 @@ def main():
...
     feature_parser.add_argument('--no-cached-namespace', dest="cached_namespace", action='store_false',
                                 help="Don't use the cached namespace.")
-    feature_parser.add_argument('--severity', dest="severity", type=int, help="Report anything with the given severity or higher.")
-    # todo check that it is between 0 and 10
-    parser.set_defaults(cached_namespace=True)
-    parser.set_defaults(severity=10)
+    feature_parser.add_argument('--severity', dest="severity", type=int,
+                                help="Report anything with the given severity or higher as error (default: %(default)s).",
```
Comment (rly) [inline src/pynwb/validate.py:56]:
```suggestion
                                help="Report anything with the given severity or higher as errors (default: %(default)).",
```

**Review by rly — CHANGES_REQUESTED** [review]

File: src/pynwb/validate.py (hunk 1 above)
Comment (t-b) [reply src/pynwb/validate.py:52]: Did you see https://docs.python.org/3/library/argparse.html?highlight=argumentparser#help? This suggests that the trailing `s` is indeed correct.

The help output also looks correct.

```
  --cached-namespace    Use the cached namespace (default: True).
```

Comment (rly) [reply src/pynwb/validate.py:52]: My bad! I thought it was a typo and should have checked first.

**Comment by bendichter** [comment]
@rly and @t-b what's the status of this?

**Comment by t-b** [comment]
I need to find time and resources to work on this. Hopefully in 2021. Feel free to close if you don't want to have it open that long without action.

**Comment by bendichter** [comment]
That's fine. Response time on issues/PRs has become a core metric for us, so I'm sifting through "unresponded issues."

**Comment by t-b** [comment]
Already solved via https://github.com/hdmf-dev/hdmf/pull/542.

**Pull request closed without merging**